.github/workflows/unit-test-on-pull-request.yml

```diff
@@ -23,83 +21,101 @@ jobs:
       - name: Linter
         run: |
           go version
-          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.56.2
-          make lint
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh \
+            | sh -s -- -b $(go env GOPATH)/bin v1.56.2
+          make lint TARGET_ARCH=${{ matrix.target_arch }}
 
   test:
-    name: Test
+    name: Test (${{ matrix.target_arch }})
     runs-on: ubuntu-24.04
     strategy:
-      fail-fast: true
-      max-parallel: 2
       matrix:
-        go: ["stable"]
+        target_arch: [amd64, arm64]
     steps:
       - name: Clone code
         uses: actions/checkout@v4
       - name: Set up environment
         uses: ./.github/workflows/env
       - name: Build
-        run: |
-          echo $PATH
-          make test
+        run: make test TARGET_ARCH=${{ matrix.target_arch }}
       - name: Tests
-        run: |
-          make test
+        run: make test
```

Comment:
Is it on purpose that you run the tests 2x? One with TARGET_ARCH and another one with the host architecture?

Reply:
Nope -- good catch!

Reply:
Looks like this is an old copy paste fail. This is the corresponding code on `main`:

https://github.com/elastic/otel-profiling-agent/blob/18a7e9f442d9d7ae2212f42e7c2a02c9e78af5df/.github/workflows/unit-test-on-pull-request.yml#L42-L48

I fixed it now.